docs: add scenario doc for Backup and Restore of OCI Artifacts and Images (#1778)
oras-project/oras: docs/proposals/backup-restore-scenarios.md

* Duplication of blobs wastes storage and network bandwidth.

The result is frustrated DevOps engineers, wasted resources, and security gaps that erode confidence in artifact management processes. There is a clear need for an integrated, reliable, and user-friendly solution to:

* Efficiently acquire and promote artifacts across local environment and registries.
* Create portable, structured backups in standard formats.
* Restore registry state, including all metadata and dependencies.
* Empower teams to meet security, compliance, and operational requirements across air-gapped, multi-cloud, and hybrid environments.

## Scenarios

This document illustrates real-world scenarios highlighting these challenges and how unified, structured backup and restore functionality built into `oras` can significantly improve user experience, operational efficiency, and supply chain security.

### Scenario 1: Creating Offline Snapshots for Air-Gapped Environments

Dave, a security engineer at a FinTech company. To create a snapshot of the image and its referrers in an air-gapped environment, Dave needs to run the following flow:

> 1. Packages the image and its referrers from an OCI image layout into a `.tar` for portability.

Author: This change is not correct as it is supposed to use third-person here.

Recently, a misconfigured cleanup job accidentally deleted an entire production repository. Although image builds could be recreated, many old tags still in use by customers were lost, along with their associated attestations. Old tags are particularly important for her team. For example, base images like `ubuntu` has multiple old tags `18.04`, `20.04`, etc, many enterprise customers still run old versions and those tags must remain accessible for hotfix builds, compliance audits, or rollback. The team spent hours manually recovering what they could using `docker` and `oras`, and scattered scripts, but couldn’t fully restore the original state.

To prevent future incidents and meet compliance audit requirements, Alice needs a unifed, portable solution to back up an entire repository including all tags and referrers into a single archive or directory. She also wants to restore it quickly and completely if something goes wrong.

> With a holistic backup and restore solution in `oras`, Alice can capture the full state of a repository and recover it with confidence, improving operational resilience and audit readiness while eliminating complex manual work.

Reviewer: ```suggestion
With a holistic backup and restore solution in `oras`, Alice can capture all artifacts of a repository and recover it with confidence, improving operational resilience and audit readiness while eliminating complex manual work.
```

### Scenario 1: Creating Offline Snapshots for Air-Gapped Environments

Dave is a security engineer at a FinTech company that operates in a highly regulated environment. As part of their infrastructure security policies, the production environment is fully air-gapped to reduce the attack surface and comply with regulatory standards.

To deploy software to this air-gapped environment, Dave is responsible for preparing offline snapshots of all required container images and associated artifacts, including SBOMs, signatures, and attestations. These artifacts must be reviewed, signed, and bundled in a secure, verifiable manner before being transferred to production.

The process typically involves:

1. Packaging an image and all its linked referrers from a remote registry into a single `.tar` archive or directory using OCI image layout format locally.
2. Transferring the compressed snapshot over secure channels to the air-gapped network.

> 3. Restoring the image and referrer rtifacts into an internal OCI-compliant registry for deployment.

Reviewer 2: typo:

```suggestion
3. Restoring the image and referrer artifacts into an internal OCI-compliant registry for deployment.
```

> Today, Dave has no native way to do this using `docker save/load`, `oras pull/push`, or other common tools. The lack of a unified snapshot solution means that artifacts must be copied manually or scripted with `oras copy`, often resulting in incomplete transfers (e.g., missing signatures or dependencies), blobs duplicated across files and no assurance of artifact integrity and completeness, difficulty validating artifact integrity upon restore. See an example GitHub issue [#730](https://github.com/oras-project/oras/issues/730) for details.

Reviewer 2: > no assurance of artifact integrity and completeness, difficulty validating artifact integrity upon restore

Why?

Author: To validate artifact integrity, users need to check the supply chain artifacts such as signature, SBOM, etc. As mentioned above, the current solution (docker save/load or scripts with `oras copy`) may result in incomplete transfers (e.g., missing signatures or dependencies) which makes validation process hard.
